amp_conf/htdocs/admin/libraries/BMO/Ajax.class.php

```diff
@@ -63,14 +63,15 @@ public function doRequest($module = null, $command = null) {
 			$this->ajaxError(403, 'ajaxRequest declined');
 		}
 
+		$url = parse_url($_SERVER['HTTP_REFERER']);
```

Comment:
It can be spoofed, but because of the previous check you will only get here both referrer and host are 127.0.0.1: this requires the request to have gone to 127.0.0.1 _and_ for it also to have set referrer to 127.0.0.1. The second is easy to spoof, but the first? And by checking referrer, we add a little twist in, by requiring the request to be well formed both for host an referrer.